gpt_engineer/data/file_repository.py

```diff
@@ -197,20 +244,93 @@ def __delitem__(self, key: Union[str, Path]) -> None:
         elif item_path.is_dir():
             shutil.rmtree(item_path)
 
+    def _xml_tree_supported_files(self, directory: Path, parent: ET.Element) -> bool:
+        valid_extensions = {
+            ext for lang in supported_languages for ext in lang["extensions"]
+        }
+
+        has_supported_files = False
+        subdir_element = ET.Element("directory", {"name": directory.name})
+
+        for item in sorted(directory.iterdir()):
+            if item.is_dir():
+                if self._xml_tree_supported_files(item, subdir_element):
+                    has_supported_files = True
+            elif item.suffix in valid_extensions:
+                ET.SubElement(subdir_element, "file", {"name": item.name})
+                has_supported_files = True
+
+        if has_supported_files:
+            parent.append(subdir_element)
+            return True
+
+        return False
+
+    def _xml_tree_all_files(self, directory: Path, parent: ET.Element):
+        subdir_element = ET.SubElement(parent, "directory", {"name": directory.name})
+
+        for item in sorted(directory.iterdir()):
+            if item.is_dir():
+                self._xml_tree_all_files(item, subdir_element)
+            else:
+                ET.SubElement(subdir_element, "file", {"name": item.name})
+
+    def _supported_files(self, directory: Path) -> str:
+        valid_extensions = {
+            ext for lang in supported_languages for ext in lang["extensions"]
+        }
+        file_paths = [
+            str(item)
+            for item in sorted(directory.rglob("*"))
+            if item.is_file() and item.suffix in valid_extensions
+        ]
+        return "\n".join(file_paths)
+
+    def _all_files(self, directory: Path) -> str:
+        file_paths = [
+            str(item) for item in sorted(directory.rglob("*")) if item.is_file()
+        ]
+        return "\n".join(file_paths)
+
+    def to_path_xml_string(self, supported_code_files_only: bool = False) -> str:
```

Comment:
Why use xml instead of the shorter yaml?

Reply:
I've removed this now as it wasn't in use